horizon_bsn/api/rest/neutron.py

```diff
@@ -0,0 +1,261 @@
+# NETWORK TEMPLATE ASSIGNMENT
+##################################################################
+
+
+@urls.register
+class HeatStack(generic.View):
+    """API for Router_get"""
+    url_regex = r'heat/stack/(?P<stack_id>[^/]+|default)/$'
+
+    @rest_utils.ajax()
+    def delete(self, request, stack_id):
+        result = heat.stack_delete(request, stack_id)
+        return result
+
+    @rest_utils.ajax()
+    def get(self, request, stack_id):
+        result = heat.stack_get(request, stack_id)
+        return result.stack_status
+
+
+@urls.register
+class HeatStacks(generic.View):
+    """API for Router_get"""
+    url_regex = r'heat/stack/$'
+
+    @rest_utils.ajax()
+    def post(self, request):
+        result = heat.stack_create(request, **request.DATA)
+        return result
+
+
+@urls.register
+class HeatTemplateValidate(generic.View):
+    """API for BSN Neutron Network Template Assignment"""
+    url_regex = r'heat/templatevalidate/$'
+
+    @rest_utils.ajax()
+    def post(self, request):
+        result = heat.template_validate(request, template=request.DATA['body'])
+        return result
+
+
+@urls.register
+class NetworkTemplateAssignment(generic.View):
+    """API for BSN Neutron Network Template Assignment"""
+    url_regex = \
+        r'neutron/networktemplateassignment/(?P<assignment_id>[^/]+|default)/$'
+
+    @rest_utils.ajax()
+    def delete(self, request, assignment_id):
+        result = bsnneutron.\
+            networktemplateassignment_delete(request, assignment_id)
+        return result
+
+
+@urls.register
+class NetworkTemplateAssignments(generic.View):
+    """API for BSN Neutron Network Template Assignment"""
+    url_regex = r'neutron/networktemplateassignment/$'
+
+    @rest_utils.ajax()
+    def post(self, request):
+        result = bsnneutron.\
+            networktemplateassignment_create(request, **request.DATA)
+        return result
+
+    @rest_utils.ajax()
+    def patch(self, request):
+        result = bsnneutron.\
+            networktemplateassignment_update(
+                request,
+                request.user.project_id,
+                **{'stack_id': request.DATA['stack_id']})
+        return result
+
+    @rest_utils.ajax()
+    def get(self, request):
+        try:
+            topology = get_stack_topology(request)
+            if not topology.get('assign'):
+                return []
+            tabledata = {
+                'id': topology['template'].id,
+                'name': topology['template'].name,
+                'stack_id': topology['stack'].id,
+                'heat_stack_name': topology['stack'].stack_name,
+                'description': topology['stack'].description,
+                'status': topology['stack'].status,
+                'stack_status': topology['stack'].stack_status,
+                'stack_status_reason': topology['stack'].stack_status_reason,
+                'resources': mark_safe('<br>'.join([
```

Comment:
also, this is probably why you saw `\n` when you tried to display the network template/resource. this replaces it with `<br>`